cli/src/lockfiles/javascript.rs

```diff
@@ -91,16 +91,24 @@ impl Parseable for YarnLock {
             let resolution = package
                 .get(&"resolution".into())
                 .and_then(YamlValue::as_str)
+                .filter(|s| !s.is_empty())
                 .ok_or_else(|| "Failed to parse resolution field in yarn lock file".to_owned())?;
 
             // Ignore workspace-local dependencies like project itself ("project@workspace:.").
-            if resolution.contains("@workspace:") {
+            if resolution[1..].contains("@workspace:") {
```

Comment:
Decided it would be a good idea to exclude the `@` at the beginning of package scopes here, to ensure we're only looking at the yarn data. According to some quick searches online, it seems like this is the only place where `@` is allowed in the package name.